src/vi/mimihentai/src/eu/kanade/tachiyomi/extension/vi/mimihentai/MiMiHentai.kt

```diff
@@ -0,0 +1,201 @@
+            }
+        }
+    }
+
+    override fun imageUrlParse(response: Response): String = ""
+
+    override fun mangaDetailsRequest(manga: SManga): Request {
+        val id = manga.url.substringAfterLast("/")
+        val url = apiUrl.toHttpUrl().newBuilder()
+            .addPathSegment("info")
+            .addPathSegment(id)
+            .build()
+        return GET(url, headers)
+    }
+
+    override fun mangaDetailsParse(response: Response): SManga {
+        val res = response.parseAs<Manga>()
+        return res.let { manga ->
+            SManga.create().apply {
+                description = manga.description
+                author = manga.authors.joinToString { i -> i.name }
+                genre = manga.genres.joinToString { i -> i.name }
+                thumbnail_url = manga.coverUrl
+                title = manga.title
+            }
+        }
+    }
+
+    override fun pageListParse(response: Response): List<Page> {
+        val json = JSONObject(response.body.string())
+        val pagesArray = json.getJSONArray("pages")
+        val listUrl = (0 until pagesArray.length()).map { i ->
+            pagesArray.getString(i)
+        }
+        return listUrl.mapIndexed { index, url ->
+            Page(index, imageUrl = url)
+        }
+    }
+
+    override fun popularMangaParse(response: Response): MangasPage = latestUpdatesParse(response)
+
+    override fun popularMangaRequest(page: Int): Request = GET(
+        apiUrl.toHttpUrl().newBuilder().apply {
+            addPathSegments("tatcatruyen")
+            addQueryParameter("page", page.toString())
+            addQueryParameter("sort", "views")
+        }.build(),
+        headers,
+    )
+
+    override fun fetchSearchManga(page: Int, query: String, filters: FilterList): Observable<MangasPage> {
+        return when {
+            query.startsWith(PREFIX_ID_SEARCH) -> {
+                val id = query.removePrefix(PREFIX_ID_SEARCH)
+                client.newCall(searchMangaByIdRequest(id))
+                    .asObservableSuccess()
+                    .map { response -> searchMangaByIdParse(response, id) }
+            }
+            query.toIntOrNull() != null -> {
+                client.newCall(searchMangaByIdRequest(query))
+                    .asObservableSuccess()
+                    .map { response -> searchMangaByIdParse(response, query) }
+            }
+            else -> super.fetchSearchManga(page, query, filters)
+        }
+    }
+    private fun searchMangaByIdRequest(id: String) = GET("$apiUrl/info/$id", headers)
+
+    override fun searchMangaParse(response: Response): MangasPage = popularMangaParse(response)
+
+    private fun searchMangaByIdParse(response: Response, id: String): MangasPage {
+        val details = mangaDetailsParse(response)
+        details.url = "/info/$id"
+        client.newCall(GET("$apiUrl/gallery/$id", headers)).asObservableSuccess().map { resp -> chapterListParse(resp) }
```

Comment:
What's this line for?

Reply:
> What's this line for?

Search manga with id -> mangaDetailsParse
GET chapter -> chapterListParse

Reply:
Click to see the code around. Your return value is `MangasPage(listOf(details), false)`. There's nothing you've taken from this `chapterListParse()`. You're doing it for nothing.

Reply:
> Click to see the code around. Your return value is `MangasPage(listOf(details), false)`. There's nothing you've taken from this `chapterListParse()`. You're doing it for nothing.

ok i delete it